core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-results-seo-tool/dot-results-seo-tool.component.spec.ts

```diff
@@ -102,18 +100,38 @@ describe('DotResultsSeoToolComponent', () => {
     });
 
     it('should display mobile size for the preview', () => {
-        const previews = spectator.queryAll(byTestId('seo-preview'));
+        spectator.setInput({
+            seoMedia: SEO_MEDIA_TYPES.GOOGLE
+        });
+        spectator.detectChanges();
+        const previews = spectator.queryAll(byTestId('seo-preview-mobile'));
         expect(previews[1]).toHaveClass('results-seo-tool__version--small');
     });
 
-    it('should filter seo results by seoMedia on changes', () => {
-        spectator.fixture.componentInstance.seoMedia = 'facebook';
-        spectator.component.ngOnChanges();
-        spectator.component.currentResults.subscribe((items) => {
+    it('should filter seo results by Facebook, seoMedia on changes', () => {
+        spectator.setInput({
+            seoMedia: SEO_MEDIA_TYPES.FACEBOOK
+        });
+        spectator.detectChanges();
+        spectator.component.currentResults$.subscribe((items) => {
             expect(items.length).toEqual(3);
             expect(items[0].key).toEqual(seoOGTagsResultMock[1].key);
             expect(items[1].key).toEqual(seoOGTagsResultMock[3].key);
             expect(items[2].key).toEqual(seoOGTagsResultMock[4].key);
         });
```

Comment:
Can you break this test? I think we had several problems in templates and experiments with this type of expect. If you cannot break it, you need to use the `done()` callback to make it work